src/server/game/Entities/Unit/StatSystem.cpp

```diff
@@ -298,6 +292,10 @@ void Player::UpdateMaxHealth()
     value *= GetModifierValue(unitMod, TOTAL_PCT);
 
     SetMaxHealth((uint32)value);
+
+    Pet* pet = GetPet();
+    if (pet)
+        pet->UpdateMaxHealth();
```

Comment:
```c++
     if (Pet* pet = GetPet())
         pet->UpdateMaxHealth();
```

?

Reply:
Pets get allways Health according to owner.
Example:

Hunter HP = 10000
Pet HP = 7000 (70%)

Hunter equips now or gets buffed 
Hunter HP = 15000
Pet HP = 10500 (79%)

Reply:
it's inside
`Player::UpdateMaxHealth()`

Reply:
yep